linera-faucet/server/src/lib.rs

```diff
@@ -118,17 +123,57 @@ where
     C: ClientContext,
 {
     async fn do_claim(&self, owner: AccountOwner) -> Result<ClaimOutcome, Error> {
-        let client = self.context.lock().await.make_chain_client(self.chain_id)?;
+        let main_client = self
+            .context
+            .lock()
+            .await
+            .make_chain_client(self.main_chain_id)?;
+        let maybe_tmp_chain_id = *self.tmp_chain_id.lock().await;
+        let main_balance = main_client.local_balance().await?;
+        let tmp_chain_id = match maybe_tmp_chain_id {
```

Comment:
I think renaming to `faucet_chain_id` will make things clearer. Right now the `tmp_` prefix makes it look like it's just created and soon will be killed, which is confusing even more b/c down below we try to close it.